### PR TITLE
Reduce session server lookups

### LINKED ISSUE #478: Bedrock players' skin disappears on Java players' side on join
When a bedrock player joins the server with a linked account, their skin shows as one of the default skins on other Java players' side. This issue exists until they re-join the backend server where the bedrock players at:

https://github.com/GeyserMC/Floodgate/assets/55654482/20dc9736-f8a1-4cc4-a45b-aa7112c2e7b4

Floodgate configuration: https://paste.gg/p/anonymous/a4187c14cf3245a485a5fb3b81e048d5
Geyser dump: https://dump.geysermc.org/wXG9KCTHVd6PtmsgGfwx5SxcraXnLYae

Floodgate is installed both on the proxy and the backend, and the backend server doesn't have Geyser installed.

### LINKED ISSUE #508: Excessive session server lookups (velocity & potential other platforms)
Floodgate currently forwards a "broken" gameprofile to velocity that does not contain any properties at all, causing paper instances to register this player as not having any textures and therefor fetching mojang's authservers which causes (invalid) api calls every time a bedrock player switches a server on velocity.

I've recently contributed a patch to paper to prevent them from looking up profiles when it's not needed: 
https://github.com/PaperMC/Paper/commit/21d91ce634481f58a8e94f6c02651dbaaeb4408f

After closely monitoring this and building my own system around it I noticed this only happens to bedrock players so I dug deeper and found out this a fundamental flaw within floodgate.

Even default minecraft java accounts that have never selected a skin still have the "textures" property in their gameprofile, either Steve or Alex. I suggest floodgate applies either one of these by default, will be pushing a PR for this, however would like some more feedback on this.

Steve: eyJ0aW1lc3RhbXAiOjE1MzIwOTIwMDAsInByb2ZpbGVJZCI6InBsYXllci11dWlkIiwicHJvZmlsZU5hbWUiOiJwbGF5ZXItbmFtZSIsInRleHR1cmVzIjp7IkZJTkFMIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvYjgyN2ViZmY3ZjJlZDVjM2I5ZTdiYzZhNTE5NjkzNjdmMzljYjMwZjk3YTc0NjViOWU5OGZkNDY5NGZmODAwZiJ9fX19

Alex: eyJ0aW1lc3RhbXAiOjE1MzIwOTIwMDAsInByb2ZpbGVJZCI6InBsYXllci11dWlkIiwicHJvZmlsZU5hbWUiOiJwbGF5ZXItbmFtZSIsInRleHR1cmVzIjp7IkZJTkFMIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvYTkxMjFkMjZjNzU2NzA2YzY3MTcyZjM3YzYxNzFlZDk4NmE2NDgzODc5MjlhODc0MDQ1OGY0MTE2YjYxIn19fQ==

### PR DESCRIPTION
**Pull request by bridgelol**
Fixes #508, fixes #478.

### CODE REVIEW

**Comment by bridgelol** [comment]
Unsure if there's a better way than simply just sending an empty signature over.

**Comment by bridgelol** [comment]
This should solve this bungee & spigot-side as well (bungee doesn't need modifying as it doesn't have modern forwarding like velocity, fixing spigot also fixes bungee)

**Review by Kas-tle — APPROVED** [review]

File: velocity/src/main/java/org/geysermc/floodgate/listener/VelocityListener.java [hunk 1]
```diff
@@ -69,26 +70,21 @@ public final class VelocityListener {
         INITIAL_MINECRAFT_CONNECTION = getFieldOfType(initialConnection, minecraftConnection);
 
         // Since Velocity 3.1.0
-        Class<?> loginInboundConnection =
```
Comment (Kas-tle) [inline velocity/src/main/java/org/geysermc/floodgate/listener/VelocityListener.java:72]: Ideally please remove the formatting changes so the diff is clean and only contains actual code changes.

**Comment by onebeastchris** [comment]
https://github.com/GeyserMC/Floodgate/blob/master/bungee%2Fsrc%2Fmain%2Fjava%2Forg%2Fgeysermc%2Ffloodgate%2Flistener%2FBungeeListener.java#L134

https://github.com/GeyserMC/Floodgate/blob/master/spigot%2Fsrc%2Fmain%2Fjava%2Forg%2Fgeysermc%2Ffloodgate%2Flistener%2FPaperProfileListener.java
There are also these two occurrences where we're currently setting an empty skin - maybe it is worth it to change these too?

**Review by Tim203 — CHANGES_REQUESTED** [review]
One functional thing to keep in mind is that with this PR you always prevent the skin lookup, while you still want to do a skin lookup when the player is linked.
When a player is linked it acts like it's the linked Java player (including the same UUID etc.), meaning that it should also have the skin of the Java player. Which is something we don't store, and has to be looked up like normal.

File: velocity/src/main/java/org/geysermc/floodgate/listener/VelocityListener.java (hunk 1 above)
Comment (bridgelol) [reply velocity/src/main/java/org/geysermc/floodgate/listener/VelocityListener.java:72]: My bad, must've done this while getting rid of imports that weren't used*

**Comment by bridgelol** [comment]
> https://github.com/GeyserMC/Floodgate/blob/master/bungee%2Fsrc%2Fmain%2Fjava%2Forg%2Fgeysermc%2Ffloodgate%2Flistener%2FBungeeListener.java#L134
> 
> https://github.com/GeyserMC/Floodgate/blob/master/spigot%2Fsrc%2Fmain%2Fjava%2Forg%2Fgeysermc%2Ffloodgate%2Flistener%2FPaperProfileListener.java There are also these two occurrences where we're currently setting an empty skin - maybe it is worth it to change these too? Just for consistency

Yep, believe this would be ideal. https://github.com/GeyserMC/Floodgate/pull/509/commits/33fd6ac3e86addf1909fbcad19f24b7291021fda

**Comment by bridgelol** [comment]
> One functional thing to keep in mind is that with this PR you always prevent the skin lookup, while you still want to do a skin lookup when the player is linked. When a player is linked it acts like it's the linked Java player (including the same UUID etc.), meaning that it should also have the skin of the Java player. Which is something we don't store, and has to be looked up like normal.

Would you recommend not adding the properties if the player is linked?

**Comment by bridgelol** [comment]
> > One functional thing to keep in mind is that with this PR you always prevent the skin lookup, while you still want to do a skin lookup when the player is linked. When a player is linked it acts like it's the linked Java player (including the same UUID etc.), meaning that it should also have the skin of the Java player. Which is something we don't store, and has to be looked up like normal.
> 
> Would you recommend not adding the properties if the player is linked?

Or, alternatively, look them up ourselves?

**Comment by onebeastchris** [comment]
If the player is linked, the Java server will see a valid Java username/uuid, and should look up textures for that player

File: spigot/src/main/java/org/geysermc/floodgate/addon/data/SpigotDataHandler.java [hunk 2]
```diff
@@ -68,10 +77,10 @@ protected Object setHostname(Object handshakePacket, String hostname) throws Ill
...
             try {
                 Object[] components = new Object[]{
-                    ClassNames.HANDSHAKE_PROTOCOL.get(handshakePacket),
-                    hostname,
-                    ClassNames.HANDSHAKE_PORT.get(handshakePacket),
-                    ClassNames.HANDSHAKE_INTENTION.get(handshakePacket)
+                        ClassNames.HANDSHAKE_PROTOCOL.get(handshakePacket),
+                        hostname,
+                        ClassNames.HANDSHAKE_PORT.get(handshakePacket),
+                        ClassNames.HANDSHAKE_INTENTION.get(handshakePacket)
```
Comment (onebeastchris) [inline spigot/src/main/java/org/geysermc/floodgate/addon/data/SpigotDataHandler.java:83]: please revert formatting change

**Comment by bridgelol** [comment]
> If the player is linked, the Java server will see a valid Java username/uuid, and should look up textures for that player

I believe we should be doing skin lookups on velocity ourselves then, having spigot handle it if floodgate's ran purely on spigot is fine, however you don't want spigot instances doing a new lookup every time a player switches server on velocity.

**Comment by bridgelol** [comment]
> If the player is linked, the Java server will see a valid Java username/uuid, and should look up textures for that player

Should be good to merge now

**Review by onebeastchris — APPROVED** [review]

**Review by Tim203 — APPROVED** [review]

**Pull request merged**